lib/cocoapods/sandbox/path_list.rb

```diff
@@ -44,6 +44,32 @@ def dirs
         @dirs
       end
 
+      # List all files and directories under {dir}. This method could be executed recursively.
+      #
+      # @param  [Pathname] dir directory to traverse
+      #
+      # @return [Array<Array<String>>] The array of directories and
+      #         files in {dir}. The paths are relative from
+      #
+      def list_files_under(dir, basepath = nil)
+        dirs = []
+        files = []
+        basepath = dir.realpath unless basepath
+        dir.children.each do |path|
+          if path.directory?
+            dirs << path.to_s
+            if !path.symlink? || path.realpath.relative_path_from(basepath).to_s.start_with?('..')
```

Comment:
using `relative_path_from` instead of the string slicing here will be _incredibly_ expensive